packages/gafl-webapp-service/src/pages/renewals/renewal-start-date/route.js

```diff
@@ -1,25 +1,76 @@
 import { ADVANCED_PURCHASE_MAX_DAYS, SERVICE_LOCAL_TIME } from '@defra-fish/business-rules-lib'
-import { dateFormats } from '../../../constants.js'
-import { RENEWAL_START_DATE, RENEWAL_START_VALIDATE } from '../../../uri.js'
+import { dateFormats, PAGE_STATE } from '../../../constants.js'
+import { RENEWAL_START_DATE, LICENCE_SUMMARY } from '../../../uri.js'
 import pageRoute from '../../../routes/page-route.js'
 import Joi from 'joi'
 import JoiDate from '@hapi/joi-date'
 import moment from 'moment-timezone'
-import { displayExpiryDate } from '../../../processors/date-and-time-display.js'
+import { displayExpiryDate, cacheDateFormat } from '../../../processors/date-and-time-display.js'
 import { addLanguageCodeToUri } from '../../../processors/uri-helper.js'
+import { errorShimm } from '../../../handlers/page-handler.js'
+import { licenceToStart } from '../../licence-details/licence-to-start/update-transaction.js'
+import { ageConcessionHelper } from '../../../processors/concession-helper.js'
+
 const JoiX = Joi.extend(JoiDate)
 
-const validator = payload => {
+export const validator = (payload, options) => {
+  const { permission } = options.context.app.request
+  const endDateMoment = moment.utc(permission.renewedEndDate).tz(SERVICE_LOCAL_TIME)
   const licenceStartDate = `${payload['licence-start-date-year']}-${payload['licence-start-date-month']}-${payload['licence-start-date-day']}`
-  Joi.assert(
+
+  return Joi.assert(
     { 'licence-start-date': licenceStartDate },
     Joi.object({
-      'licence-start-date': JoiX.date().format(dateFormats).required()
-    })
+      'licence-start-date': JoiX.date()
+        .format(dateFormats)
+        .min(endDateMoment.clone().startOf('day'))
+        .max(endDateMoment.clone().add(ADVANCED_PURCHASE_MAX_DAYS, 'days'))
+        .required()
+    }).options({ abortEarly: false, allowUnknown: true })
   )
 }
 
-const getData = async request => {
+export const setLicenceStartDateAndTime = async (result, request) => {
+  const permission = await request.cache().helpers.transaction.getCurrentPermission()
+  const { payload } = await request.cache().helpers.page.getCurrentPermission(RENEWAL_START_DATE.page)
+  const endDateMoment = moment.utc(permission.renewedEndDate).tz(SERVICE_LOCAL_TIME)
+
+  if (result.error) {
+    await request.cache().helpers.page.setCurrentPermission(RENEWAL_START_DATE.page, { payload, error: errorShimm(result.error) })
+    await request
+      .cache()
+      .helpers.status.setCurrentPermission({ [RENEWAL_START_DATE.page]: PAGE_STATE.error, currentPage: RENEWAL_START_DATE.page })
+    return addLanguageCodeToUri(request, RENEWAL_START_DATE.uri)
+  }
```

Comment:
Thanks for removing the else. Does the function get invoked when the validation fails?